airflow/operators/sensors.py

```diff
@@ -207,6 +207,50 @@ def poke(self, context):
         return count
 
 
+class NamedHivePartitionSensor(BaseSensorOperator):
+    """
+    Waits for a partition to show up in Hive. Only accepts plain partition
+    names (eg dt=xxx) for the partition argument and no filters.
+
+    :param table: The name of the table to wait for, supports the dot
+        notation (my_database.my_table)
+    :type table: string
+    :param partition: The partition clause to wait for. This is passed as
+        is to the Metastore Thrift client "get_partitions_by_name" method.
+    :type partition: string
+    :param metastore_conn_id: reference to the metastore thrift service
+        connection id
+    :type metastore_conn_id: str
+    """
+    def __init__(
```

Comment:
if an array of names is passed, would we just join them with slashes? `get_partition_by_name` only takes a single name argument (https://github.com/apache/hive/blob/master/metastore/if/hive_metastore.thrift#L1140)